updated jacoco to 0.8.2 for java 10+ support

TorstenKruse opened the pull request.

ganncamp committed:
Drop SonarSource organization (#65)

CCFenner committed:
align config property name (#69)

CCFenner committed:
remove duplicate log entry (#75)

already logged at JaCoCoReportReader.java#L61

CCFenner committed:
remove misleading log entry (#76)

TobiX committed:
Adjust version to 1.6-SNAPSHOT

TobiX committed:
Update to SonarQube 6.7 (LTS)

CI build also runs against 7.3 and 7.5 to validate compatibility.

This closes #86 and fixes #72.

TobiX committed:
Update JaCoCo to 0.8.2 for Java 10+ support

Closes #84, fixes #71

TobiX committed:
Add SonarCloud integration

TobiX committed:
Add myself to developers

TobiX committed:
Minor dependency updates

TobiX committed:
Update parent and license headers

TobiX committed:
Record coverage with JaCoCo

TobiX committed:
[travis] Clone complete history for SonarCloud

TobiX committed:
Update badges and version overview

TobiX committed:
Clarify preferred code style & add fmt-maven-plugin

TobiX committed:
Remove deprecated GroovyFileSystem.sourceFiles

TobiX committed:
Stream files from SonarQube to CodeNarc

This gets rid of some deprecated API usage and allows running tests
without writing source files to the file system

TobiX committed:
Add AppVeyor configuration

TobiX committed:
Add AppVeyor badge to README

Hydragyrum committed:
Replace deprecated RulesProfile for scanner-side operations (#4)

TobiX committed:
Update project location in Maven pom.xml

TobiX committed:
Integrate codenarc converter into the default build

This helps keeping that code running and in sync with the main plugin.

TobiX committed:
Fix travis builds on external branches

TobiX committed:
Remove usage of MutableTestCase & MutableTestPlan

TobiX committed:
Use 'Paths.get(".")' for current directory in tests

TobiX committed:
Some simple fixes for deprecation warnings

TobiX committed:
Replace mocks with temporary directories

TobiX committed:
Replace mocked ActiveRules with real classes

TobiX committed:
Create a wrapper around ActiveRulesBuilder

This is a horrible workaround to be able to run tests against multiple
SonarQube versions (6.7-7.6) with incompatible testing classes...

TobiX committed:
Include CodeNarc as a git submodule

This allows us to run the converter tests on Travis-CI and should help
external contributors to understand the converter better.

TobiX committed:
Add SonarQube 7.7 to Travis-CI test matrix

TobiX committed:
Remove coupling metrics

Since SonarQube doesn't use those metrics anymore (they are on each
package and that feature was removed from SonarQube), we don't need to
collect them anymore.

TobiX committed:
Fix project description

CCFenner committed:
update data used in marketplace (#7)

TobiX committed:
"Ignore" source directories in cobertura reports

Cobertura doesn't keep the full path to analyzed files in its XML
reports, only a list of source directories which were used to find those
files. Before this change, the plugin tried to reconstruct the absolute
path from those informations and tries to "ask" SonarQube about each of
them. Now, we just try to find any input file matching the relative path
of the current file. This makes the logic much simpler and gives us a
slight chance to detect multiple files matching the same relative path.

As a side-effect, we don't need to mock the whole file system in the
test cases.

TobiX committed:
Only report overall JaCoCo coverage

Since SonarQube doesn't differentiate between unit and integration test
coverage anymore, it doesn't make any sense to keep them separate here.

TobiX committed:
Travis-CI: Tests now run on all configurations

TobiX committed:
SONAR-11064: Drop unused metric "comment_lines_data"

TobiX committed:
Repair GMetrics metrics test

TobiX committed:
Remove old metrics (fixes #6)

TobiX committed:
Enable JaCoCo coverage on all modules

TobiX committed:
Only run SonarQube analysis on one Travis-CI matrix configuration

TobiX committed:
Travis-CI: Install step is redundant

TobiX committed:
Clean up logging in codenarc converter

TobiX committed:
Fix Travis-CI URL

TobiX committed:
Make RuleParameter immutable (using AutoValue)

TobiX committed:
Update CodeNarc to 0.26.0

TobiX committed:
Run the CodeNarc converter on each build

TobiX committed:
Update CodeNarc to 0.27.0

TobiX committed:
Update submodules in AppVeyor

TobiX committed:
Make sure the submodule really was checked out

TobiX committed:
Fix minor issues (XXE in local tool)

TobiX committed:
Completly exclude groovy-ant

TobiX committed:
Fix typo

TobiX committed:
Get rid of RemediationEffortExtractor

Since I have no idea how the input file came about, this tool is
basically useless.

TobiX committed:
Remove obsolete dependencyManagement for groovy-ant

TobiX committed:
Update CodeNarc and GMetrics to 1.0

TobiX committed:
Sort RuleSet enum

TobiX committed:
Sort rules.xml the same way as the RuleSet enum

TobiX committed:
Update CodeNarc to 1.1

TobiX committed:
Add "support" for SonarQube 7.8

This would be much easier if SonarSource wouldn't drop testing APIs all
the time :(

TobiX committed:
Add SonarSource bintray repository

This allows us to use versions which haven't made it to Maven Central
yet.

TobiX committed:
Update CodeNarc to 1.2

TobiX committed:
Update CodeNarc to 1.3

TobiX committed:
Update CodeNarc to 1.4

TobiX committed:
Improve rule descriptions

- Don't swallow lists
- Make sure paragraph content doesn't contain unescaped HTML "tags"
- Represent different syntax with different HTML tags

TobiX committed:
Simpler (and correcter) paragraph parsing

This showed even more code where the old way was just silently
swallowing lines from the rule description :/

TobiX committed:
Merge pull request #13 from Inform-Software/new-codenarc

Update CodeNarc to 1.4

TobiX committed:
Minor README updates

TobiX committed:
Add changelog

TobiX committed:
Fix diff ranges in changelog

TobiX committed:
Update parent and dependencies

TobiX committed:
Add support for SonarQube 7.9

TobiX committed:
Travis: Test against Java 8 & 11

TobiX committed:
Configure maven-release-plugin

TobiX committed:
[maven-release-plugin] prepare release 1.6

TobiX committed:
[maven-release-plugin] prepare for next development iteration

TobiX committed:
Update README and CHANGELOG

TobiX commented:
This (and other issues) are fixed in [release 1.6](https://community.sonarsource.com/t/new-release-sonar-groovy-plugin-1-6/11528)

TobiX committed:
Remove backward compatibility hacks

This makes the oldest supported SonarQube version 7.8.

TobiX committed:
Minor dependency updates

TobiX committed:
Drop compatibility with JaCoCo < 0.7.5

Unfortunately, some tests were still using old, unreproducible binary
JaCoCo files. These tests have been removed.

TobiX committed:
Travis: Test against SonarQube 8.0

TobiX committed:
Tell Maven to not alter paths for sub-projects

This needs a pretty recent Maven (at least 3.6.1,
see MNG-6059 for details)

TobiX committed:
AppVeyor: Update build image for recent Maven

TobiX committed:
Test against SonarQube 8.3

TobiX committed:
Disable JaCoCo sensor if JaCoCo XML exists

TobiX committed:
Remove 'test_success_density' metric (fixes #25)

TobiX committed:
Update documentation for coverage

TobiX committed:
Add distributionManagement for BinTray (see #18)

pascalporta444 committed:
JaCoCoAnalyzer: log missing file only if extension is groovy

pascalporta444 committed:
Merge branch 'cit'

# Conflicts:
#	groovy-jacoco-previous/pom.xml
